api/tasks.go

```diff
@@ -272,6 +316,41 @@ func (g *TaskGroup) Canonicalize(job *Job) {
 		g.Update.Canonicalize()
 	}
 
+	// Merge the reschedule policy from the job
+	if jr, tr := job.Reschedule != nil, g.ReschedulePolicy != nil; jr && tr {
+		jobReschedule := job.Reschedule.Copy()
+		jobReschedule.Merge(g.ReschedulePolicy)
+		g.ReschedulePolicy = jobReschedule
+	} else if jr && !job.Reschedule.Empty() {
```

Comment:
Why restrict on the jobs being empty? What if I just want to set a no restart policy at the job and have it be inherited across all task groups?

Reply:
When looking at how the update stanza worked, I found this commit https://github.com/hashicorp/nomad/commit/6723bade37513cd14d7ad9470c6a4a75d6252cfb and at the time convinced myself that reschedule also needed this. Having thought about it more and reading the related issue, I realize now that the problem it was solving was around reading the update stanza from the job struct via the api(which we don't have for reschedule) and then applying a job update which would override a legitimate non empty update stanza. 
In the case of reschedule, we don't add the update stanza to the job's API struct, its always inherited from the job but persisted at the task group so you are right we can remove this restriction. Will fix!